lld.mixin

```diff
@@ -0,0 +1,11 @@
+{
+    "build": {
+        "lld": {
+            "cmake-args": [
+                "-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld",
+                "-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld",
+                "-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld"
```

Comment:
Unless there's a reason to omit the argument for static linking.. I'd suggest

```suggestion
                "-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld",
                "-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=lld",
                "-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld"
                "-DCMAKE_STATIC_LINKER_FLAGS=-fuse-ld=lld"
```

Reply:
Static linking uses the program `ar` and setting that will cause errors.

Reply:
Nevermind then.

Reply:
@tylerjw what is the behavior of static linking without this then? Should ar conventionally be installed with lld?

Reply:
`ar` is part of the gnu compiler tools.  afaik clang also uses `ar` for static linking.

Reply:
I misspoke.  Clang provides its own version of ar: https://llvm.org/docs/CommandGuide/llvm-ar.html

In either case, this is not something we have to configure, it is the default behavior.

Reply:
When the compiler does static linking it calls the version of `ar` that is part of its toolchain.  We don't want to override that behavior because the linkers ld, lld, gold, mold, etc do not replace ar.